be/src/storage/delta_writer.cpp

```diff
@@ -219,10 +219,14 @@ Status DeltaWriter::_init() {
                            sort_key_idxes.begin(), sort_key_idxes.end())) {
             _partial_schema_with_sort_key = true;
         }
-        writer_context.tablet_schema = writer_context.partial_update_tablet_schema.get();
+        _new_tablet_schema = writer_context.partial_update_tablet_schema;
+        writer_context.tablet_schema = _new_tablet_schema.get();
         writer_context.partial_update_mode = _opt.partial_update_mode;
     } else {
-        writer_context.tablet_schema = &_tablet->tablet_schema();
+        /// copy schema
+        _new_tablet_schema = _tablet->tablet_schema().deep_copy();
```

Comment:
You can just copy json column. For another columns, you can use the pointer or reference instead.